### PR TITLE
DB-720 : ps 5.6 Can not create a clustering index on a partitioned to…

### PR DESCRIPTION
**Pull request by george-lorch**
…kudb table

During an attempt to create a clustering key, Percona Server has an extra check where it queries the storage engine handlerton to see if it supports clustered keys.
When a partitioned table is in use, the partition storage engine is answering this query rather than the TokuDB storage engine.

We have to consider that even though Percona Server does not currently support a mixed set of storage engines for partitions, there is much code that is built around that possibility.

This fix adds the HTON_SUPPORTS_CLUSTERED_KEYS flag to the partition handlerton to indicate that it _could_ support clustered keys if all of the underlying handlertons do.

"mysql_prepare_create_table" has been changed to enumerate and all partitions if they exist ant test each partitions real handlerton for the HTON_SUPPORTS_CLUSTERED_KEYS

Removed some vestigial TokuDB handler method.

Re-enabled the tokudb.cluster_key_part. test and added another test case to it.

Updated the main.tokudb_clustering_key_grammar to include some partitioning grammar tests.

### CODE REVIEW

**Comment by george-lorch** [comment]
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1051/

This is based on the GCA determined commit of 2071aeef64c3f9c4f59636e3690ce1fd48d0529a

File: mysql-test/suite/tokudb/t/cluster_key_part.test [hunk 1]
```diff
@@ -5,19 +5,25 @@ source include/have_partition.inc;
 set default_storage_engine='tokudb';
```
Comment (laurynas-biveinis) [inline mysql-test/suite/tokudb/t/cluster_key_part.test:5]: IMHO it needs to be tested not only that DDL goes through, but that clustered indexes can actually be updated and data read back from them successfully. I have doubts whether that's in the scope of the current MP (do unpartitioned TokuDB tables have such tests?), but it has to be at least tried somehow before we ship the feature.

Comment (george-lorch) [inline mysql-test/suite/tokudb/t/cluster_key_part.test:5]: Not a bad idea and another point where TokuDB might have some mtr weakness. Scopewise, I don't really care, if we can make a better test when we touch a feature, we should and automatically assume it is within the scope, especially with a feature like this where we are opening up an entire new area (well, new to PS) that hasn't been executed at all due to lifting this limitation.

I believe there is a cluster_key_something test that I can snip a few lines from and append to this test to validate that it is functioning beneath the partition handler.

**Comment by george-lorch** [comment]
Expanded tests, re-recorded and push --force the branch.

**Comment by george-lorch** [comment]
jenkins not re-run due to current backlog in running jobs, full mtr run locally and completes as expected. If jenkins clears up before merging, I'll fire another job.

**Comment by george-lorch** [comment]
jenkins running : http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/1055/

**Comment by laurynas-biveinis** [comment]
LGTM

**Comment by george-lorch** [comment]
Rebased to new 5.6 GCA and new jenkins http://jenkins.percona.com/job/percona-server-5.6-param/1068/

**Pull request merged**